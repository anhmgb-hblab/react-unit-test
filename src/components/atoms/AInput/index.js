import React from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';
import { EyeInvisibleOutlined, EyeTwoTone } from '@ant-design/icons';
import StyledAInput from './styled';

const { Password, TextArea } = Input;

const AInput = ({
  placeholder,
  onChange,
  disabled,
  readonly,
  size,
  type,
  autoSize,
  allowClear,
  value,
  onFocus,
  label,
}) => {
  const renderLabel = () => {
    if (label === '') return null;
    return <div className="input__label">{label}</div>;
  };

  const renderInput = () => {
    switch (type) {
      case 'text': return (
        <Input
          placeholder={placeholder}
          onChange={onChange}
          disabled={disabled}
          readOnly={readonly}
          size={size}
          allowClear={allowClear}
          value={value}
          onFocus={onFocus}
        />
      );
      case 'password': return (
        <Password
          placeholder={placeholder}
          onChange={onChange}
          disabled={disabled}
          readOnly={readonly}
          size={size}
          allowClear={allowClear}
          value={value}
          onFocus={onFocus}
          iconRender={(visible) => (visible ? <EyeTwoTone /> : <EyeInvisibleOutlined />)}
        />
      );
      case 'textarea': return (
        <TextArea
          placeholder={placeholder}
          onChange={onChange}
          disabled={disabled}
          readOnly={readonly}
          size={size}
          autoSize={autoSize}
          allowClear={allowClear}
          value={value}
          onFocus={onFocus}
        />
      );
      default: {
        throw new Error(`Unhandled type: ${type}`);
      }
    }
  };

  return (
    <StyledAInput>
      {renderLabel()}
      {renderInput()}
    </StyledAInput>
  );
};

export default AInput;

AInput.defaultProps = {
  placeholder: '',
  onChange: () => {},
  disabled: false,
  readonly: false,
  size: 'middle',
  type: 'text',
  autoSize: false,
  allowClear: false,
  value: '',
  onFocus: () => {},
  label: '',
};

AInput.propTypes = {
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  readonly: PropTypes.bool,
  size: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  autoSize: PropTypes.bool || PropTypes.object,
  allowClear: PropTypes.bool,
  onFocus: PropTypes.func,
  label: PropTypes.string,
};
