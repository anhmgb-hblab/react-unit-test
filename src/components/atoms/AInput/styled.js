import styled from 'styled-components';

export default styled.div`
  .input__label {
    margin-bottom: 8px;
  }
`;
