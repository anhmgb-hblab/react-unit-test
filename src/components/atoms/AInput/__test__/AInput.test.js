import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import AInput from '../index';

describe('Test AInput running correctly', () => {
  it('test button props', () => {
    const wrapper = mount(
      <AInput
        label="Email"
        value="email"
        placeholder="Input email here"
        size="middle"
        type="text"
      />,
    );

    expect(wrapper.props().size).toEqual('middle');
    expect(wrapper.props().type).toEqual('text');
    expect(wrapper.props().label).toEqual('Email');
    expect(wrapper.props().value).toEqual('email');
    expect(wrapper.props().placeholder).toEqual('Input email here');
    expect(wrapper.props().disabled).toEqual(false);
    expect(wrapper.props().autoSize).toEqual(false);
    expect(wrapper.props().allowClear).toEqual(false);
    expect(wrapper.props().readonly).toEqual(false);
  });

  it('test input function', () => {
    const onFocus = jest.fn(() => 'focused');

    const wrapper = mount(
      <AInput
        label="Email"
        value="email"
        onFocus={onFocus}
      />,
    );

    wrapper.props().onFocus();
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus.mock.results[0].value).toBe('focused');
  });

  it('test rendering', () => {
    const tree = mount(
      <AInput
        label="Email"
        value="email"
        placeholder="Input email here"
        size="middle"
        type="text"
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
