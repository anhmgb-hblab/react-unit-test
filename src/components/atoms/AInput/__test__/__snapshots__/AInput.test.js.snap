// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AInput running correctly test rendering 1`] = `
<AInput
  allowClear={false}
  autoSize={false}
  disabled={false}
  label="Email"
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="Input email here"
  readonly={false}
  size="middle"
  type="text"
  value="email"
>
  <styled.div>
    <div
      className="sc-bdfBQB fLvfYJ"
    >
      <div
        className="input__label"
      >
        Email
      </div>
      <Input
        allowClear={false}
        disabled={false}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Input email here"
        readOnly={false}
        size="middle"
        type="text"
        value="email"
      >
        <ClearableLabeledInput
          allowClear={false}
          bordered={true}
          disabled={false}
          element={
            <input
              className="ant-input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Input email here"
              readOnly={false}
              type="text"
              value="email"
            />
          }
          focused={false}
          handleReset={[Function]}
          inputType="input"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Input email here"
          prefixCls="ant-input"
          readOnly={false}
          size="middle"
          triggerFocus={[Function]}
          type="text"
          value="email"
        >
          <input
            className="ant-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Input email here"
            readOnly={false}
            type="text"
            value="email"
          />
        </ClearableLabeledInput>
      </Input>
    </div>
  </styled.div>
</AInput>
`;
